pkg/upgrade/drain_manager_test.go

```diff
@@ -53,6 +53,37 @@ var _ = Describe("DrainManager tests", func() {
 		Expect(err).To(Succeed())
 		Expect(node.Spec.Unschedulable).To(BeTrue())
 	})
+	It("DrainManager should drain all nodes it receives", func() {
+		ctx := context.TODO()
+
+		node1 := createNode("node1")
+		node2 := createNode("node2")
+		node3 := createNode("node3")
+
+		drainManager := upgrade.NewDrainManager(k8sInterface, upgrade.NewNodeUpgradeStateProvider(k8sClient, log), log)
+		drainSpec := &DrainSpec{
+			Enable:         true,
+			Force:          false,
+			PodSelector:    "",
+			TimeoutSecond:  1,
+			DeleteEmptyDir: true,
+		}
+		nodeArray := []*corev1.Node{node1, node2, node3}
+		err := drainManager.ScheduleNodesDrain(ctx, &upgrade.DrainConfiguration{Nodes: nodeArray, Spec: drainSpec})
+		Expect(err).To(Succeed())
+
+		time.Sleep(time.Second)
+
+		err = k8sClient.Get(ctx, types.NamespacedName{Name: node1.Name}, node1)
```

Comment:
can you use a new var for the observed node ? e.g `observedNode` instead of reusing node1 ?

Reply:
Thanks, addressed